plugins/API/API.php

```diff
@@ -484,7 +485,21 @@ public function getBulkRequest($urls)
 
         $result = array();
         foreach ($urls as $url) {
-            $req = new Request($url . '&format=php&serialize=0');
+            $params = Request::getRequestArrayFromString($url . '&format=php&serialize=0');
+
+            if (isset($params['urls']) && $params['urls'] == $urls) {
+                // by default 'urls' is added to $params as Request::getRequestArrayFromString adds all $_GET/$_POST
+                // default parameters
+                unset($params['urls']);
+            }
+
+            if (!empty($params['segment']) && strpos($url, 'segment=') > -1) {
+                // only unsanitize input when segment is actually present in URL, not when it was used from
+                // $defaultRequest in Request::getRequestArrayFromString from $_GET/$_POST
+                $params['segment'] = urlencode(Common::unsanitizeInputValue($params['segment']));
```

Comment:
The `urlencode` is not really needed I believe but added it for now. Otherwise we pass eg `actions>5` (instead of  `actions%3E5`) to the API methods. This shouldn't be a problem as all output should be proper encoded anyway. I did it because a comment here https://github.com/piwik/piwik/blob/2.15.1-b10/core/API/Request.php#L484 says we want URL encoded segments. I know `Piwik\Segment` can handle both urlencoded and urldecoded so it might be good to remove the `urlencode` here as it might be a problem if the segment parameter in the bulk request URL was encoded twice already.